src/stats.rs

```diff
@@ -553,7 +514,7 @@ impl<T: StatisticsLogFormatter> StatsLoggerBuilder<T> {
     }
 
     /// Construct the StatisticsLogger - this will start the interval logging if requested.
-    pub fn fuse(self, logger: slog::Logger) -> StatisticsLogger<T> {
+    pub fn fuse<T: StatisticsLogFormatter>(self, logger: slog::Logger) -> StatisticsLogger {
```

Comment:
Even this bound is too much, this isn't needed unless (a) you've enabled the `interval_logging` feature and (b) you called `with_interval_log` on the builder.

I think we should rejig the builder on the back of this (maybe there's two `fuse` functions?  maybe `with_interval_log` is parameterized?  maybe the builder still is?)

Reply:
Seems reasonable. I'll propose a design.

Reply:
Hopefully the new code looks better.

Reply:
Yes, it looks much better to me, thanks.